Include access logs in audit user report

gherceg opened the pull request:
## Product Description
<!-- Where applicable, describe user-facing effects and include screenshots. -->
<img width="1549" height="458" alt="image" src="https://github.com/user-attachments/assets/42244b73-2900-4f34-885d-1c660eea95d8" />

## Technical Summary
<!--
    Provide a link to any tickets, design documents, and/or technical specifications
    associated with this change. Describe the rationale and design decisions.
-->
Currently it isn't clear to support what information is included in audit logs since we only show navigation logs in the report. This makes it impossible to see events like failed login attempts.

This change makes it so that the report support can view shares the same data as what is included in an auditcare export. This way support can actually use this report as a point of reference for what to expect if they proceed with a full export.

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->
This change makes it so that more code is shared between this audit user report and the function to export audit logs. Both of these are internal tools that could potentially be shared with users upon request, but otherwise is not an actively used public feature.

As far as the subtle change to default to an empty string if `domain` is None, the python csv lib already treats None as an empty string when writing data, as [these docs](https://docs.python.org/3/library/csv.html#csv.writer) say:
> To make it as easy as possible to interface with modules which implement the DB API, the value [None](https://docs.python.org/3/library/constants.html#None) is written as the empty string


### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->
No

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

gherceg committed:
Organize imports

gherceg commented:
If filtering by domain, this change is basically useless. I don't think it hurts to add it, though maybe it is worth adding a filter to include/exclude access and navigation events. That way access events won't pollute a report where we just want to see navigation logs for example.

jingcheng16 commented:
Can the first commit be split into smaller size? Like "rename xxx to xxx" so it is easier to review?

gherceg committed:
Share code to obtain row for log event

This is used by management commands like export_audit_data currently, so
sharing it with the user audit report makes sense.

The csv writer already converts values of None to empty strings,
so moving the "domain or ''" logic into shared code is safe.

gherceg committed:
Show both navigation and audit events in report

The report is still limited to a max number of records, so
sorting by date after combining access and navigation audit events
doesn't feel too terrible.

gherceg commented:
> Can the first commit be split into smaller size? Like "rename xxx to xxx" so it is easier to review?

Split into what I thought made sense.

jingcheng16 reviewed (approved):
Thank you for making the changes! Not blocking but agree to Daniel's feedback.

jingcheng16 commented on corehq/apps/hqadmin/reports.py:
```diff
@@ -151,23 +152,19 @@ def rows(self):
     @memoized
     def _is_limit_exceeded(self):
-        where = filters_for_navigation_event_query(
+        where = filters_for_audit_event_query(
```
Just realized this filter is only applied to `NavigationEventAudit` in this function...

millerdev commented on corehq/apps/auditcare/utils/export.py:
```diff
@@ -120,16 +136,21 @@ def write_generic_log_event(writer, event):
         action = event.access_type
```
@gherceg do you plan to follow up with a separate PR for this?

gherceg replied:
I believe @jingcheng16 is working on this as part of a related task (correct me if I'm wrong!)

jingcheng16 replied:
I plan to add a filter for the `action` column, and if the action belongs to `AccessAudit`, we only query `AccessAudit`, otherwise, we will just query `NavigationAudit` and use it as a baseline to infer if the returned result will excess the limit, given `AccessAudit` will be much less than `NavigationAudit`

gherceg replied:
And just to be clear, you also plan to change the action code to a human readable string?
